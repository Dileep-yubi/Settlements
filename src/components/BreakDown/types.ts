export type Payment = {
  title: string;
  value: string;
  currency?: string;
};
export type BreakDownProps = {
  summary: Array<Payment>;
};
