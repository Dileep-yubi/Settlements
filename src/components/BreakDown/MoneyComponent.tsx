import React from "react";

type Amount = {
  currency: string;
  value: number;
};
export type MoneyComponentProps = {
  amount: Amount;
};
export const MoneyComponent: React.FC<MoneyComponentProps> = ({
  amount,
}: MoneyComponentProps) => {
  if (!amount.currency || amount.value == null) {
    return <></>;
  }
  return (
    <>
      {amount.currency} {amount.value}
    </>
  );
};
