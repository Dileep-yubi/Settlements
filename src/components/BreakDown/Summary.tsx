import React from "react";
import styled from "styled-components";
import { Payment } from "./types";

export type FlexDivProps = {
  flexDirection: string;
  justifyContent: string;
  alignItems: string;
};
export const FlexDiv = styled.div`
  width: 100%;
  display: flex;
  flex-direction: "row";
  justify-content: "space-between";
  align-items: "center";
  gap: 5px;
  line-height: 26px;
  padding: 15px;
`;

const Summary: React.FC<Payment> = ({
  title,
  value,
  currency,
}: Payment) => {
  return (
    <FlexDiv>
      <span>{title}</span>
      <span>{`${currency + " " || ""} ${value}`}</span>
    </FlexDiv>
  );
};

export default Summary;
