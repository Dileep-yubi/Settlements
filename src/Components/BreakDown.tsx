import React from "react";
import { PlanSummary } from "./EmiPlanComponent/EmiPlanComponent";
import { Divider } from "./StyledComponents";
import Summary from "./Summary";

export interface BreakDownProps {
  summary: Array<PlanSummary>;
  divider?: boolean;
}

export const BreakDown = (props: BreakDownProps) => {
  return (
    <>
      {props.summary.map((summery: PlanSummary, i: number) => {
        return (
          <div key={i} style={{ fontSize: "14px", fontWeight: "normal" }}>
            <Summary summery={summery} />
            {props.divider && props.summary.length - 1 === i && <Divider />}
          </div>
        );
      })}
    </>
  );
};
