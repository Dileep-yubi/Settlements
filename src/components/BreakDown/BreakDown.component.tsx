import React from "react";
import { Divider } from "../BreakDownCard/BreakDownCard.component";
import Summary from "./Summary";
import { BreakDownProps } from "./types";

const BreakDown: React.FC<BreakDownProps> = (props: BreakDownProps) => {
  return (
    <>
      {props.summary.map((e: any, i: any) => {
        return (
          <div key={i} style={{ fontSize: "14px", fontWeight: "normal" }}>
            {i !== 0 && <Divider data-testid="divider" />}
            <Summary title={e.title} value={e.value} currency={e.currency} />
          </div>
        );
      })}
    </>
  );
};

export default BreakDown;
