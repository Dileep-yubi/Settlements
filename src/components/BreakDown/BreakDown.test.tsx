import { render, screen } from "@testing-library/react";
import BreakDown from "./BreakDown.component";

describe("BreakDown", () => {
  const renderComponent = () =>
    render(
      <BreakDown
        summary={[
          { title: "EMI", value: "100", currency: "USD" },
          { title: "Interest", value: "1", currency: "USD" },
        ]}
      />
    );
  it("snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
    expect(screen.getByText("EMI")).toBeInTheDocument();
    expect(screen.getByText("USD 100")).toBeInTheDocument();
    expect(screen.getByTestId("divider")).toBeInTheDocument();
  });
});
