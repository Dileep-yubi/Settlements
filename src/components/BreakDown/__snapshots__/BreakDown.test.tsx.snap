// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreakDown snapshot 1`] = `
<div>
  <div
    style="font-size: 14px; font-weight: normal;"
  >
    <div
      class="sc-gswNZR pwObR"
    >
      <span>
        EMI
      </span>
      <span>
        USD  100
      </span>
    </div>
  </div>
  <div
    style="font-size: 14px; font-weight: normal;"
  >
    <div
      class="sc-bcXHqe iXsvJl"
      data-testid="divider"
    />
    <div
      class="sc-gswNZR pwObR"
    >
      <span>
        Interest
      </span>
      <span>
        USD  1
      </span>
    </div>
  </div>
</div>
`;
